src/charm.py

```diff
@@ -72,7 +78,8 @@ def __init__(self, *args: typing.Any):
         self.certificates = TLSCertificatesRequiresV3(self, TLS_CERT_RELATION)
         self._tls = TLSRelationService(self.model, self.certificates)
         self._ingress_provider = IngressPerAppProvider(charm=self, relation_name=INGRESS_RELATION)
-        self.http_provider = HTTPProvider(self, REVERSE_PROXY_RELATION)
+        self.http_requirer = HTTPRequirer(self, REVERSE_PROXY_RELATION)
+        self.http_provider = HTTPProvider(self, WEBSITE_RELATION)
```

Comment:
As discussed on MM I'd rather call these variables  "website_provider" and "reverseproxy_requirer"

Reply:
Got it, I changed the name of these attributes to  `website_provider` and `reverseproxy_requirer`